context/app/static/js/hooks/useSearchData.ts

```diff
@@ -0,0 +1,339 @@
+
+interface UseHitsData<Documents> {
+  searchHits: NonNullable<Required<SearchHit<Documents>>[]>;
+  isLoading: boolean;
+}
+
+const defaultConfig = {
+  useDefaultQuery: false,
+  fetcher: fetch,
+  fallbackData: {},
+};
+
+export default function useSearchData<Documents, Aggs>(
+  query: SearchRequest,
+  {
+    useDefaultQuery = defaultConfig.useDefaultQuery,
+    fetcher = defaultConfig.fetcher,
+    ...swrConfig
+  }: UseSearchDataConfig | undefined = defaultConfig,
+): UseSearchData<Documents, Aggs> {
+  const requestInit = useRequestInit({ query, useDefaultQuery });
+  const { elasticsearchEndpoint } = useAppContext();
+
+  const { data, isLoading } = useSWR<SearchResponseBody<Documents, Aggs>>(
+    { query, requestInit, url: elasticsearchEndpoint },
+    fetcher,
+    swrConfig,
+  );
+
+  // The data is guaranteed to be defined because we provide a fallbackData
+  return { searchData: data!, isLoading };
+}
+
+export function useSearchHits<Documents>(
+  query: SearchRequest,
+  {
+    useDefaultQuery = defaultConfig.useDefaultQuery,
+    fetcher = defaultConfig.fetcher,
+    ...swrConfig
+  }: UseSearchDataConfig | undefined = defaultConfig,
+): UseHitsData<Documents> {
+  const { searchData, isLoading } = useSearchData(query, { useDefaultQuery, fetcher, ...swrConfig });
+  const searchHits = (searchData?.hits?.hits ?? []) as Required<SearchHit<Documents>>[];
+  return { searchHits, isLoading };
+}
+
+// Multisearch
+
+type PluralQuery<T extends { query: SearchRequest }> = Omit<T, 'query'> & { queries: SearchRequest[] };
+
+function useRequestInits({ queries, useDefaultQuery }: PluralQuery<RequestInitArgs>): RequestInit[] {
+  const authHeader = useAuthHeader();
+  const inits = useMemo<RequestInit[]>(
+    () =>
+      queries.map((query) => {
+        const body = createSearchRequestBody({ query, useDefaultQuery });
+        return buildSearchRequestInit({ body, authHeader });
+      }),
+    [queries, useDefaultQuery, authHeader],
+  );
+  return inits;
+}
+
+const defaultMultiFetchConfig = {
+  useDefaultQuery: false,
+  fetcher: multiFetch,
+  fallbackData: [],
+};
+
+interface UseMultiSearchData<Documents, Aggs = Record<string, AggregationsAggregate>> {
+  searchData: SearchResponseBody<Documents, Aggs>[] | undefined;
+  isLoading: boolean;
+}
+export function useMultiSearchData<Documents, Aggs>(
+  queries: SearchRequest[],
+  { useDefaultQuery = false, fetcher = multiFetch, ...swrConfig }: UseSearchDataConfig = defaultMultiFetchConfig,
+): UseMultiSearchData<Documents, Aggs> {
+  const requestInits = useRequestInits({ queries, useDefaultQuery });
+  const { elasticsearchEndpoint } = useAppContext();
+
+  const { data: searchData, isLoading } = useSWR<SearchResponseBody<Documents, Aggs>[]>(
+    { queries, requestInits, url: elasticsearchEndpoint },
+    fetcher,
+    swrConfig,
+  );
+
+  return { searchData, isLoading };
+}
+
+// Infinite search
+
+function getTotalHitsCount(results?: SearchResponseBody<unknown, unknown>) {
```

Comment:
Are we using <unknown, unknown> here and below because the generic doesn't affect the function's return type?